### PR TITLE
Allow grain UI to show more than 2 digits past the decimal

### PR DESCRIPTION
**Pull request by magwalk**
Allows instance maintainers to customize the number of decimals
that are displayed with their currency (fix for ticket #2778) 

# Description
Adds a new property `decimalsToDisplay` to the `currencyDetails.json` config. The default value is 2. 

Sample `currencyDetails.json`:
```
{
  "decimalsToDisplay": 4
}
```

Adding this property results in this change to the UI: 
Screen Shot 2021-04-05 at 6 01 45 PM

# Test Plan
- load the UI without a currencyDetails.json file
- load the UI with a currencyDetails.json file with a custom value for `decimalsToDisplay`: 
```
{
  "decimalsToDisplay": 4
}
```
- load the UI with a currencyDetails.json file that has other values but not `decimalsToDisplay`, like so: 
```
{
  "currencyName": "honey",
  "currencySuffix": "h" 
}
```

### CODE REVIEW

File: src/api/currencyConfig.js [hunk 1]
```diff
@@ -4,11 +4,12 @@ import * as C from "../util/combo";
 import * as NullUtil from "../util/null";
 
 /**
- * Shape of concurrencyDetails.json on disk
+ * Shape of currencyDetails.json on disk
```
Comment (magwalk) [inline src/api/currencyConfig.js:7]: typo fix

**Comment by magwalk** [comment]
I think I got all the places where currency is displayed and should show decimal points, let me know if there are others I didn't catch.

**Comment by blueridger** [comment]
We should probably apply this to the stats circle on the explorer home as well. And also the GrainReceiptTable component of the LedgerViewer.

**Comment by KuraFire** [comment]
@magwalk YAY! Thank you for this much-wanted improvement :)

**Design thought:**
Correct me if I'm wrong, but this seems to create a (brand?) new config file, `currencyDetails.json`? Can we make configuration values go in a central configuration location? 

(I know Dandelion had hesitations about that, but I'm totally unclear on why we would want there to be 5 places in your setup to configure one single instance to your needs.)

**For my UI review:**
Does this come with an upper limit? We have no need for letting people enter "144" and break the UI. I don't know if we need to support more than 8 digits?

**Comment by blueridger** [comment]
@KuraFire currencyDetails.json is an existing file, this just adds a new attribute to that file.

We only support up to 18 digits. Adding a conditional throw in the upgrade function could be nice to enforce the value into 0-18.

**Comment by magwalk** [comment]
So this brings up an interesting breaking point for the stats circles UI. With the current text size on the stats circles, there is a limit of 9 digits that can be displayed before the text breaks out of the circles. A question I have for @KuraFire - how should the UI respond when there are more than 9 digits? 

Here's how it breaks (this example shows 10 digits): 
Screen Shot 2021-04-07 at 8 07 24 AM

**Comment by magwalk** [comment]
> So this brings up an interesting breaking point for the stats circles UI. With the current text size on the stats circles, there is a limit of 9 digits that can be displayed before the text breaks out of the circles. A question I have for @KuraFire - how should the UI respond when there are more than 9 digits?
> 

Talked it over with a couple devs and we are going to push off this work because this seems to be an edge case that isn't likely to be broken by any of our users. I will create a bug issue to track the work to fix this.

**Review by BrianBelhumeur — APPROVED** [review]

File: src/api/currencyConfig.js [hunk 2]
```diff
@@ -32,17 +35,26 @@ function upgrade(c: SerializedCurrencyDetails): CurrencyDetails {
   return {
```
Comment (blueridger) [inline src/api/currencyConfig.js:35]:
```suggestion
  if (c.decimalsToDisplay > 18 || c.decimalsToDisplay < 0) throw new Error("currencyConfig: decimalsToDisplay must be between 0-18")
  return {
```

**Comment by magwalk** [comment]
New loom demonstrating the new stats circles values with compact formatting https://www.loom.com/share/26c674259ab94adf9970bb19220221e5

**Review by KuraFire — APPROVED** [review]
Awesome work! ❤️

**Review by BrianBelhumeur — APPROVED** [review]

**Pull request merged**